Submitted By: Jim Gifford < jim at cross-lfs dot org>
Date: 2009-02-08
Upstream Status: Unknown
Origin: Joe Ciccone 
Description: This patch gives the ability to change the libdir.
             Updated for Iproute2-2.6.28 by Jim Gifford
             Discovered tc directory is text files moved to /usr/share

diff -Naur iproute2-2.6.28.orig/include/iptables.h iproute2-2.6.28/include/iptables.h
--- iproute2-2.6.28.orig/include/iptables.h	2009-01-15 12:25:04.000000000 -0800
+++ iproute2-2.6.28/include/iptables.h	2009-02-08 16:52:27.636947898 -0800
@@ -4,8 +4,12 @@
 #include "iptables_common.h"
 #include "libiptc/libiptc.h"
 
+#ifndef LIBDIR
+#define LIBDIR "/lib"
+#endif
+
 #ifndef IPT_LIB_DIR
-#define IPT_LIB_DIR "/usr/local/lib/iptables"
+#define IPT_LIB_DIR LIBDIR "/iptables"
 #endif
 
 #ifndef IPPROTO_SCTP
diff -Naur iproute2-2.6.28.orig/Makefile iproute2-2.6.28/Makefile
--- iproute2-2.6.28.orig/Makefile	2009-01-15 12:25:04.000000000 -0800
+++ iproute2-2.6.28/Makefile	2009-02-08 16:54:40.912668809 -0800
@@ -4,6 +4,7 @@
 CONFDIR=/etc/iproute2
 DOCDIR=/share/doc/iproute2
 MANDIR=/share/man
+SHAREDIR=/usr/share
 
 # Path to db_185.h include
 DBM_INCLUDE:=/usr/include
diff -Naur iproute2-2.6.28.orig/netem/Makefile iproute2-2.6.28/netem/Makefile
--- iproute2-2.6.28.orig/netem/Makefile	2009-01-15 12:25:04.000000000 -0800
+++ iproute2-2.6.28/netem/Makefile	2009-02-08 16:52:27.636947898 -0800
@@ -20,9 +20,9 @@
 	$(HOSTCC) $(CCOPTS) -I../include -o $@ $@.c -lm
 
 install: all
-	mkdir -p $(DESTDIR)/lib/tc
+	mkdir -p $(DESTDIR)$(SHAREDIR)/tc
 	for i in $(DISTDATA); \
-	do install -m 755 $$i $(DESTDIR)/lib/tc; \
+	do install -m 755 $$i $(DESTDIR)$(SHAREDIR)/tc; \
 	done
 
 clean:
diff -Naur iproute2-2.6.28.orig/tc/Makefile iproute2-2.6.28/tc/Makefile
--- iproute2-2.6.28.orig/tc/Makefile	2009-01-15 12:25:04.000000000 -0800
+++ iproute2-2.6.28/tc/Makefile	2009-02-08 16:54:21.971717000 -0800
@@ -75,10 +75,10 @@
 	$(AR) rcs $@ $(TCLIB)
 
 install: all
-	mkdir -p $(DESTDIR)$(LIBDIR)/tc
+	mkdir -p $(DESTDIR)$(SHAREDIR)/tc
 	install -m 0755 tc $(DESTDIR)$(SBINDIR)
 	for i in $(TCSO); \
-	do install -m 755 $$i $(DESTDIR)$(LIBDIR)/tc; \
+	do install -m 755 $$i $(DESTDIR)$(SHAREDIR)/tc; \
 	done
 
 clean:
diff -Naur iproute2-2.6.28.orig/tc/tc_util.c iproute2-2.6.28/tc/tc_util.c
--- iproute2-2.6.28.orig/tc/tc_util.c	2009-01-15 12:25:04.000000000 -0800
+++ iproute2-2.6.28/tc/tc_util.c	2009-02-08 16:56:15.030906272 -0800
@@ -24,8 +24,8 @@
 #include "utils.h"
 #include "tc_util.h"
 
-#ifndef LIBDIR
-#define LIBDIR "/usr/lib/"
+#ifndef SHAREDIR
+#define SHAREDIR "/usr/share"
 #endif
 
 const char *get_tc_lib(void)
@@ -34,7 +34,7 @@
 
 	lib_dir = getenv("TC_LIB_DIR");
 	if (!lib_dir)
-		lib_dir = LIBDIR "/tc/";
+		lib_dir = SHAREDIR "/tc";
 
 	return lib_dir;
 }
